components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -696,6 +700,30 @@ private String getCaptchaParams(String tenantDomain) {
                                 log.debug("Enforcing recaptcha for resend confirmation is not enabled.");
                             }
                         }
+                    } else if (captchaEnabledConfigName.equals(connectorConfig.getName())) {
+                        if (Boolean.parseBoolean(connectorConfig.getValue())) {
+                            Property[] maxFailedConfig = BasicAuthenticatorDataHolder.getInstance()
+                                    .getIdentityGovernanceService()
+                                    .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName}, tenantDomain);
+                            Property maxFailedProperty = maxFailedConfig[0];
+                            try {
+                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
+                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
```

Comment:
Shall we add a debug log here too, when the no of failed attempts passed the max failed attempts

Reply:
fixed fbecd0cf90e48bdbe48887baf899da44143a3f3d